qiskit/dagcircuit/_dagnode.py

```diff
@@ -0,0 +1,102 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2017, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""
+   Object to represent the information at a node in the DAGCircuit
+"""
+
+
+class DAGNode:
+    """
+    Object to represent the information at a node in the DAGCircuit
+
+    It is used as the return value from *_nodes() functions and can
+    be supplied to functions that take a node.
+
+    """
+    def __init__(self, node_id=None, data_dict=None):
+        """ Create a node """
+        self.node_id = node_id
+
+        self.data_dict = data_dict
```

Comment:
Do you think there is a meaningful semantic difference between "a DAGNode does not have a `data_dict`" and "a DAGNode has an empty `data_dict`"? If it is not a too strong difference, we could take the chance to generalize a bit and go for the second option, as in:
```
self.data_dict = data_dict or {}
```

It would allow interacting with `data_dict` in a more generic way (as in, the user would always know it is a dict, without having to check for `None` explicitly).

Reply:
No, I don't think there is really a difference. A DAGNode should always be created with a `data_dict` so I think the best solution is probably to remove the default value, as no where in the code is a DAGNode created with only an id, and I can't think of a situation where this would happen/be useful.